akvo/rsr/management/commands/report_d2d.py

```diff
@@ -6,82 +6,146 @@
         .filter(results__indicators__periods__in=periods)
         .values_list('id', flat=True)
     )
+    indicators = [
+        id_ for _, id_ in
+        filter(lambda x: x[0] == 'indicator', log_entries)
+    ]
     projects_from_indicators = set(
         Project.objects
         .filter(results__indicators__in=indicators)
         .values_list('id', flat=True)
     )
+    results = [
+        id_ for _, id_ in
+        filter(lambda x: x[0] == 'result', log_entries)
+    ]
     projects_from_results = set(
         Project.objects.filter(results__in=results)
         .values_list('id', flat=True)
     )
+    return (
+        projects_from_comments
+        | projects_from_updates
+        | projects_from_periods
+        | projects_from_indicators
+        | projects_from_results
+    )
+
+
+def get_unique_logins(date_start, date_end):
+    return LoginLog.objects.\
+        filter(success=True, created_at__range=(date_start, date_end))\
+        .values('email')\
+        .annotate(total=Count('email'))
+
 
-    results_projects = (projects_from_comments
-                        | projects_from_updates
-                        | projects_from_periods
-                        | projects_from_indicators
-                        | projects_from_results)
+def get_published_projects(date_latest):
+    return Project.objects.filter(created_at__lt=date_latest, publishingstatus__status=PublishingStatus.STATUS_PUBLISHED)
 
-    return results_projects
+
+def get_indicator_updates(date_start, date_end):
+    return IndicatorPeriodData.objects\
+        .filter(Q(created_at__range=(date_start, date_end)) | Q(last_modified_at__range=(date_start, date_end)))
+
+
+def get_orgs_reporting_to_iati(date_start, date_end):
+    return IatiExport.objects\
+        .filter(created_at__range=(date_start, date_end))\
+        .annotate(project_count=Count('projects'))\
+        .filter(project_count__gt=0)\
+        .values_list('reporting_organisation_id', flat=True)\
+        .distinct()
+
+
+def get_projects_reporting_to_iati(date_start, date_end):
+    return IatiExport.objects\
+        .filter(created_at__range=(date_start, date_end))\
+        .values_list('projects', flat=True)\
+        .distinct()
+
+
+def make_orgs_reporting_to_iati_list(orgs, date_start, date_end):
+    orgs_list = tablib.Dataset()
+    orgs_list.headers = ['name', 'long_name', 'iati_org_type', 'report_count', 'project_count']
+    for org_id in sorted(set(orgs)):
+        org = Organisation.objects.get(pk=org_id)
+        exports = IatiExport.objects.filter(
+            created_at__range=(date_start, date_end),
+            reporting_organisation=org
+        )
+        projects = set(exports.values_list('projects', flat=True))
+        if not projects:
+            continue
+        orgs_list.append([
+            org.name,
+            org.long_name,
+            org.iati_org_type_unicode(),
+            exports.count(),
+            len(projects)
+        ])
```

Comment:
This makes 3 queries per organisation. It can use a single `Organisation` queryset and loop over the results.